include/linux/kfuzztest.h

```diff
@@ -0,0 +1,449 @@
+ *
+ * Regions should be padded to respect alignment constraints of their
+ * underlying types. These padded regions are poisoned by KFuzzTest to ensure
+ * that KASAN catches OOB accesses.
+ *
+ * The format consists of three main components:
+ * 1. A reloc_region_array: Defines the memory layout of the target structure
+ *	by partitioning the payload into logical regions. Each logical region
+ *	should contain the byte representation of the type that it represents,
+ *	including any necessary padding.
+ * 2. A reloc_table: Provides "linking" instructions that tell the kernel how
+ *	to patch pointer fields to point to the correct regions. By design,
+ *	the first region (index 0) is passed as input into a FUZZ_TEST.
+ * 3. A Payload: The raw binary data for the structure and its associated
+ *	buffers. This should be aligned to the maximum alignment of all
+ *	regions to satisfy alignment requirements of the input types, but this
+ *	isn't checked by the parser. The payload should have at least 8 bytes
+ *	of trailing padding.
+ *
+ * For a detailed specification of the binary layout see the full documentation
+ * at: Documentation/dev-tools/kfuzztest.rst
+ */
+
+/**
+ * struct reloc_region - Describes a single contiguous memory region in the
+ * payload.
+ *
+ * @start: The byte offset of this region from the start of the payload, which
+ *	should be aligned to the alignment requirements of the region's
+ *	underlying type.
+ * @size: The size of this region in bytes.
+ */
+struct reloc_region {
+	uint32_t start;
+	uint32_t size;
+};
+
+/**
+ * struct reloc_region_array - An array of all memory regions.
+ * @num_regions: The total number of regions defined.
+ * @regions: A flexible array of `num_regions` region descriptors.
+ */
+struct reloc_region_array {
+	uint32_t num_regions;
+	struct reloc_region regions[];
+};
+
+/**
+ * struct reloc_entry describes a single pointer to be patched.
+ *
+ * @region_id: The index of the region in the `reloc_region_array` that
+ *	contains the pointer.
+ * @region_offset: The start offset of the pointer inside of the region.
+ * @value: contains the index of the pointee region, or KFUZZTEST_REGIONID_NULL
+ *	if the pointer is NULL.
+ */
+struct reloc_entry {
+	uint32_t region_id;
+	uint32_t region_offset;
+	uint32_t value;
+};
+
+/**
+ * struct reloc_entry is an array of all pointer relocations required by an
+ * input.
+ *
+ * @num_entries: the number of pointer relocations.
+ * @payload_offset: the number of padded bytes between the last relocation in
+ *	entries, and the start of the payload data.
+ * @entries: array of relocations.
+ */
+struct reloc_table {
+	uint32_t num_entries;
+	uint32_t payload_offset;
+	struct reloc_entry entries[];
+};
+
+int __kfuzztest_write_cb_common(struct file *filp, const char __user *buf,
+				size_t len, loff_t *off, void *arg,
+				size_t arg_size);
+
+/**
+ * __kfuzztest_parse_input validates and parses the KFuzzTest input format.
```

Comment:
For what it's worth, I think it's less common to write kernel-doc comments for non-exported functions (are these double-underscore functions meant to be exported)?

I don't have a strong preference here, especially given that you already wrote them, so let's hear what the kernel people say.